src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -573,72 +573,101 @@ query_dfs_mount(const char *path)
 	return idx;
 }
 
-static void
+/* Discover fuse mount points from /proc/self/mounts and env DAOS_MOUNT_POINT.
+ * Return 0 for success. Otherwise return Linux errno. This function interception
+ * is still effective as long as num_dfs is not zero even if discover_daos_mount()
+ * returns a non-zero value.
+ */
+static int
 discover_daos_mount(void)
 {
-	int   idx, len_fs_root;
+	int   idx, len_fs_root, rc;
 	char *fs_root   = NULL;
 	char *pool      = NULL;
 	char *container = NULL;
 
 	/* Find the list of dfuse from /proc/mounts */
-	discover_dfuse();
+	rc = discover_dfuse();
+	if (rc)
+		D_DEBUG(DB_ANY, "discover_dfuse() failed: %d (%s)\n", rc, strerror(rc));
 
 	/* Add the mount if env DAOS_MOUNT_POINT is set. */
 	fs_root = getenv("DAOS_MOUNT_POINT");
 	if (fs_root == NULL)
-		return;
+		goto out;
 
 	if (num_dfs >= MAX_DAOS_MT) {
-		D_WARN("dfs_list[] is full already. Need to incease MAX_DAOS_MT.\n");
-		return;
+		D_ERROR("dfs_list[] is full already. Need to incease MAX_DAOS_MT.\n");
+		abort();
 	}
 
 	if (access(fs_root, R_OK)) {
 		D_DEBUG(DB_ANY, "no read permission for %s: %d (%s)\n", fs_root, errno,
 			strerror(errno));
-		return;
+		if (rc == 0)
+			rc = EACCES;
+		goto out;
 	}
 
 	idx = query_dfs_mount(fs_root);
 	if (idx >= 0)
-		return;
+		goto out;
 
 	/* Not found in existing list, then append this new mount point. */
 	len_fs_root = strnlen(fs_root, DFS_MAX_PATH);
 	if (len_fs_root >= DFS_MAX_PATH) {
 		D_DEBUG(DB_ANY, "DAOS_MOUNT_POINT is too long. It is ignored.\n");
-		return;
+		if (rc == 0)
+			rc = ENAMETOOLONG;
+		goto out;
 	}
 
 	pool = getenv("DAOS_POOL");
 	if (pool == NULL) {
 		D_DEBUG(DB_ANY, "DAOS_POOL is not set.\n");
-		return;
+		if (rc == 0)
+			rc = EINVAL;
+		goto out;
 	}
 
 	container = getenv("DAOS_CONTAINER");
 	if (container == NULL) {
 		D_DEBUG(DB_ANY, "DAOS_CONTAINER is not set.\n");
-		return;
+		if (rc == 0)
+			rc = EINVAL;
+		goto out;
 	}
 
 	D_STRNDUP(dfs_list[num_dfs].fs_root, fs_root, len_fs_root);
-	if (dfs_list[num_dfs].fs_root == NULL)
-		return;
+	if (dfs_list[num_dfs].fs_root == NULL) {
+		if (rc == 0)
+			rc = ENOMEM;
+		goto out;
+	}
 
 	dfs_list[num_dfs].pool         = pool;
 	dfs_list[num_dfs].cont         = container;
 	dfs_list[num_dfs].dfs_dir_hash = NULL;
 	dfs_list[num_dfs].len_fs_root  = len_fs_root;
 	atomic_init(&dfs_list[num_dfs].inited, 0);
 	num_dfs++;
+
+out:
+	if (num_dfs == 0)
+		D_DEBUG(DB_ANY, "There is no any DFS mount point found.\n");
```

Comment:
Grammar.  "is no any", perhaps you mean "are not any"?  Instead of "mount points" perhaps "mounts"

Reply:
I'd even suggest simplifying to something like "No DFS mount points found"

Reply:
Fixed as you suggested. Thank you!